synapse/server_notices/server_notices_manager.py

```diff
@@ -159,3 +149,55 @@ def get_notice_room_for_user(self, user_id):
 
         logger.info("Created server notices room %s for %s", room_id, user_id)
         return room_id
+
+    @defer.inlineCallbacks
+    def _get_existing_notice_room_for_user(self, user_id):
+        # type: (str) -> defer.Deferred[Optional[str]]
```

Comment:
we haven't done this elsewhere and I'm not sure it's a convention I'd like to see adopted. Honestly if you feel the need for type annotations I'd go the whole hog and do a conversion to `async` on the server notices code in a separate PR.

Reply:
I don't think it's important so I'll remove it. The reason I did it that way was because PyCharm was showing me an error if I tried to do it inline.